codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -129,56 +120,60 @@ let makeGetNextPage = (
     })
   }
 
-  async (
-    ~fromBlock,
-    ~toBlock,
-    ~logger,
-    ~contractAddressMapping,
-    ~forceWildcardEvents,
-    ~isPreRegisteringDynamicContracts,
-  ) => {
-    let logSelections = try {
-      switch (forceWildcardEvents, isPreRegisteringDynamicContracts) {
-      | (true, true) => preRegWildcardLogSelection
-      | (true, false) => wildcardLogSelection
-      | (false, _) =>
-        getNormalLogSelectionOrThrow(~contractAddressMapping, ~isPreRegisteringDynamicContracts)
-      }
-    } catch {
-    | exn =>
-      exn->ErrorHandling.mkLogAndRaise(
-        ~logger,
-        ~msg="Failed getting log selection in contract interface manager",
-      )
-    }
-
-    let startFetchingBatchTimeRef = Hrtime.makeTimer()
+  let getLogSelectionOrThrow = switch selection {
+  | Wildcard({}) => {
+      let wildcardLogSelection = contracts->Belt.Array.flatMap(contract => {
+        contract.events->Belt.Array.keepMap(event => {
+          let module(Event) = event
+          let {isWildcard, topicSelections} =
+            Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+          isWildcard ? Some(LogSelection.makeOrThrow(~addresses=[], ~topicSelections)) : None
+        })
+      })
 
-    //fetch batch
-    let pageUnsafe = await Helpers.queryLogsPageWithBackoff(
-      () =>
-        queryLogsPage(
-          ~client,
-          ~fromBlock,
-          ~toBlock,
-          ~logSelections,
-          ~fieldSelection,
-          ~nonOptionalBlockFieldNames,
-          ~nonOptionalTransactionFieldNames,
-          ~logger=Logging.createChild(
-            ~params={"type": "Hypersync Query", "fromBlock": fromBlock, "serverUrl": endpointUrl},
-          ),
-        ),
-      logger,
-    )
+      let preRegWildcardLogSelection = contracts->Belt.Array.flatMap(contract => {
+        contract.events->Belt.Array.keepMap(event => {
+          let module(Event) = event
+          let {isWildcard, preRegisterDynamicContracts, topicSelections} =
+            Event.handlerRegister->Types.HandlerTypes.Register.getEventOptions
+          isWildcard && preRegisterDynamicContracts
+            ? Some(LogSelection.makeOrThrow(~addresses=[], ~topicSelections))
+            : None
+        })
+      })
 
-    let pageFetchTime =
-      startFetchingBatchTimeRef->Hrtime.timeSince->Hrtime.toMillis->Hrtime.intFromMillis
+      (~contractAddressMapping as _, ~isPreRegisteringDynamicContracts) => {
+        if isPreRegisteringDynamicContracts {
+          preRegWildcardLogSelection
+        } else {
+          wildcardLogSelection
```

Comment:
What is the difference between these two?

Reply:
`preRegWildcardLogSelection` filters out wildcard events which don't have preRegistration enabled

Reply:
Oh nvmnd, I see now 👍🏼